server/app/services/program/ProgramServiceImpl.java

```diff
@@ -210,7 +216,17 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
   }
 
   private boolean hasProgramNameCollision(String programName) {
-    return getActiveAndDraftPrograms().getProgramNames().contains(programName);
+    ImmutableSet<String> programNames = getActiveAndDraftPrograms().getProgramNames();
+
+    if (programNames.contains(programName)) {
+      return true;
+    }
+
+    Slugify slugifier = new Slugify();
+    ImmutableSet<String> programSlugs =
+        programNames.stream().map(slugifier::slugify).collect(ImmutableSet.toImmutableSet());
```

Comment:
(nit) Is there a difference between collecting as a set, and then calling contains, vs just using `Stream::anyMatch`? 

If not I think using `Stream::anyMatch` might be more clear.

Reply:
done